File: algorithms/searching/linearSearch/java/LinearSearch.java
public class LinearSearch {
    /**
     * Linear search on an array to find the index of an element.
     *
     * O(n) Time
     * @param arr The array to search.
     * @param target The target element to find.
     * @return The index of the target element if found, otherwise -1.
     */
    public int search(int[] arr, int target) {
        if (arr == null) return -1;
        for (int i = 0; i < arr.length; i++) {
            if (target == arr[i]) return i;
        }
        return -1;
    }
}

class LinearSearchUsage {
    public static void main(String[] args) {
        int[] arr = {9, 8, 7, 6, 5, 4, 3, 2, 1, 0};
  
        LinearSearch linearSearch = new LinearSearch();
        System.out.println("LinearSearch: " + linearSearch.search(arr, 4));
    }
} 

File: algorithms/searching/linearSearch/js/linearSearch.js
/**
 * Linear search on an array to find the index of an element.
 *
 * O(n) Time
 * @param {Array<number>} arr The array to search.
 * @param {number} target The target element to find.
 * @returns {number} The index of the target element if found, otherwise -1.
 */
function linearSearch(arr, target) {
  if (typeof target !== "number" || !arr) return -1;
  for (let i = 0; i < arr.length; i++) {
    if (target === arr[i]) return i;
  }
  return -1;
}

console.log("linearSearch", linearSearch([9, 8, 7, 6, 5, 4, 3, 2, 1, 0], 4));

module.exports = linearSearch;
